src/content/1.7/modules/concepts/templating/vuejs/_index.md

```diff
@@ -0,0 +1,206 @@
+---
+title: Using Vue.js
+menuTitle: Vue.js
+weight: 2
+chapter: true
+---
+
+# Using Vue.js
+
+Depending on the PrestaShop version running on the shop, either on the back or the front-office,
+at least one template renderer is available to display content on the page. With Javascript,
+it is possible to update & animate the page content without having to refresh the page.
+
+jQuery is available on PrestaShop to help implementing these interface updates.
+For complex interfaces, with state management for instance, other frontend options could be considered.
+[Vue.js](https://vuejs.org/) is a JavaScript Framework that will be helpful in this case.
+This chapter covers its integration on a module.
+
+{{% children %}}
+
+## 1- Creation of a Vue.js application in the module directories
+
+The objective of this part is to make Vue.js available in the project. 
+
+We will store the Vue.js source code in a dedicated folder that will be present in the module only
+during the development part. Running Vue.js will be possible by compiling and storing the JS files
+in the proper assets folder.
+
+During this tutorial, we use a new folder `_dev` that will be created at the root directory of the module.
+
+Installation of Vue.js can be done via [several ways](https://vuejs.org/v2/guide/installation.html),
+but we will use the CLI option to get all the stack.
+
+* Installation of Vue.js CLI tool (globally)
+
+```
+$ npm install -g @vue/cli
+```
+
+* Creation of the project
+
+```
+$ vue create _dev
+```
+
+We chose the preset with Vue 2 & Babel for this tutorial.
+
+After the initialization process with some questions and downloads, a whole standalone application
+will be created and available in the `_dev` folder. Running npm commands are defined in the `package.json`
+file to run or build it.
+
+```
+$ cd _dev
+$ npm run build
+```
+
+At the moment, we have a Vue.js application that can run independantly from the module.
+In the next parts, this standalone application will be adapted to run inside a PrestaShop page.
+## 2- Cleanup
+
+Supposing that the Vue.js workspace is the `_dev` folder:
+
+* Remove the `_dev/public/` folder.
+* In the `_dev/package.json` folder, remove the `serve` script.
+
+## 3- Customization of the application for a PrestaShop context
+
+In [the module structure]({{< ref "/1.7/modules/creation/module-file-structure.md" >}}), JavaScript
+files must be stored in `views/js/`. We will ask the compiler to write the distributable files in there.
+
+Create or edit `_dev/vue.config.js` with:
```

Comment:
Having JS/assets stuff in the views files looks so weird to me, views are designed to be :
- A view component
- Templating kinds of stuff

Reply:
That is something that could evolve with the next major version of PrestaShop, same thing for the PHP classes that could end in the `src/` folder instead of `classes/`.

But at this time, I'm sticking to the current structure we require on https://devdocs.prestashop.com/1.7/modules/creation/module-file-structure/. As that is the structure we have from the first versions of PS, that would be strange to store generated assets in another path, don't you think?

Reply:
So, the fact that we did this on Blockwishlist is due to the fact that it's a symfony module? Or it's a miss from us?

I agree with the fact that we should stick to what we defined in the past tbh, but as soon as we can change this, it would be way clearer :)

Reply:
It seems that several modules started to use their own new folder to store their assets, Faceted search has also (another folder)[https://github.com/PrestaShop/ps_facetedsearch/tree/dev/views/dist] for that matter.

The current validation rules still check the files are in `views/js` & `views/css`, so I'd advise to keep them until there's one announcement about the structure.

Reply:
Ok 👍